HTTPServerProject/HTTPServerResponse/ResponseByPath.cs

```diff
@@ -0,0 +1,155 @@
+using System;
+using System.Net;
+using System.Net.Sockets;
+using System.IO;
+using HTTPServerProject.Responses;
+using HTTPServerProject.Responses.Write;
+using HTTPServerProject.WriteStream;
+
+namespace HTTPServerProject.Path
+{
+    public class RequestPath
+    {
+        IWriteStreams writer;
+
+        public Dictionary<string, object> pathDict = new Dictionary<string, object>();
```

Comment:
Gonna highlight this here because it looks like you started down a path with `pathDict`. 

So I mentioned changing this code doesn't cause any unit tests to fail, which is a problem. Remember the 1st rule of TDD - no code without a failing unit test. 

You'll find your unit tests, at least at the start, are redundant. But the feature tests you're making pass are higher on the testing triangle, somewhere between acceptance and integration. They typically only cover positive cases, and they require the entire server running. They also don't provide much _design_ pressure, and that's where I think unit tests would help.

If you write unit tests you'll find yourself realizing are both redundant and repetitive - like the code is here, and that's a hint. You're tests are telling you "hey your code could be more generic, then you'd need less tests AND less code." I think you started to see this and that's why the pathDict is here - if every path was just data in a lookup table, then your unit tests would verify that the looking up the response from the lookup table worked, and your feature tests would validate the _specific_ cases like "method_options".  

If you move in that direction you'll probably run into a weird corner case or two but ultimately end up with code that hits both technical requirements.